fractal_server/app/api/v1/workflow.py

```diff
@@ -12,34 +12,53 @@
 """
 from fastapi import APIRouter
 from fastapi import Depends
+from fastapi import HTTPException
+from fastapi import status
+from sqlalchemy.exc import NoResultFound
+from sqlmodel import select
 
 from ...db import AsyncSession
 from ...db import get_db
 from ...security import current_active_user
 from ...security import User
 
+from ...models.workflow import Workflow
+
 router = APIRouter()
 
 
 @router.post("/")
 async def create_workflow(
+    workflow: Workflow,
     user: User = Depends(current_active_user),
     db: AsyncSession = Depends(get_db),
 ):
     """
     Add new global workflow
     """
-    raise NotImplementedError
+    # Check that there is no workflow with the same name
+    stm = select(Workflow).where(Workflow.name == workflow.name)
+    res = await db.execute(stm)
+    if res.scalars().all():
+        raise HTTPException(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            detail=f"Workflow name ({workflow.name}) already in use",
+        )
```

Comment:
this was not a requirement but you have a good point. When is it that we cannot reuse the same workflow name? I think it should be on a per project basis (please @tcompa  comment on this).

pending Tom's comment, please only check that the same name is not taken within the same project rather than globally.

Note: if we needed a unique constraint globally, we would implement it at the mapper level.

Reply:
There can exist two workflows with the same name, but only if they are associated to different projects